extensions/tokenizers/src/main/java/ai/djl/huggingface/tokenizers/HuggingFaceTokenizer.java

```diff
@@ -535,4 +480,152 @@ static PaddingStrategy fromValue(String value) {
+        public Builder optTruncateFirstOnly() {
+            options.put("truncation", TruncationStrategy.ONLY_FIRST.name());
+            return this;
+        }
+
+        /**
+         * Enables truncation to only truncate the second item.
+         *
+         * @return this builder
+         */
+        public Builder optTruncateSecondOnly() {
+            options.put("truncation", TruncationStrategy.ONLY_SECOND.name());
+            return this;
+        }
+
+        /**
+         * Enables or Disables default padding behavior for the tokenizer.
+         *
+         * @param enabled whether to enable default padding behavior
+         * @return this builder
+         */
+        public Builder optPadding(boolean enabled) {
+            options.put("padding", String.valueOf(enabled));
+            return this;
+        }
+
+        /**
+         * Enables padding to pad sequences to previously specified maxLength, or modelMaxLength if
+         * not specified.
+         *
+         * @return this builder
+         */
+        public Builder optPadToMaxLength() {
+            options.put("padding", PaddingStrategy.MAX_LENGTH.name());
+            return this;
+        }
+
+        /**
+         * Sets maxLength for padding and truncation.
+         *
+         * @param maxLength the length to truncate and/or pad sequences to
+         * @return this builder
+         */
+        public Builder optMaxLength(int maxLength) {
+            options.put("maxLength", String.valueOf(maxLength));
+            return this;
+        }
+
+        /**
+         * Sets padToMultipleOf for padding.
+         *
+         * @param padToMultipleOf the multiple of sequences should be padded to
+         * @return this builder
+         */
+        public Builder optPadToMultipleOf(int padToMultipleOf) {
+            options.put("padToMultipleOf", String.valueOf(padToMultipleOf));
+            return this;
+        }
+
+        /**
+         * Configures the builder with the arguments.
+         *
+         * @param arguments the arguments
+         */
+        public void configure(Map<String, ?> arguments) {
+            for (Map.Entry<String, ?> entry : arguments.entrySet()) {
+                options.put(entry.getKey(), entry.getValue().toString());
+            }
+        }
+
+        /**
+         * Builds the translator.
+         *
+         * @return the new translator
+         * @throws IOException when IO operation fails in loading a resource
+         */
+        public HuggingFaceTokenizer build() throws IOException {
+            String tokenizerName = options.get("tokenizer");
+            if (tokenizerName != null) {
+                return HuggingFaceTokenizer.newInstance(tokenizerName, options);
+            }
+            if (tokenizerPath == null) {
+                throw new IllegalArgumentException("Missing tokenizer path.");
+            }
+            return HuggingFaceTokenizer.newInstance(tokenizerPath, options);
+        }
```

Comment:
I think it is unlikely a user would do this, but whether they use the builder or newInstance to create a tokenizer they could set maxLength and/or padToMultiple of without explicitly enabling padding/truncation. I think there are 3 things we could do in that case:

1. Nothing, leave as is
2. Log a warning letting the user know no truncation/padding will be used even though it looks like they are trying to use it
3. Silently enable truncation/padding in this case. 

I'd prefer option 2 or 1, what do you think?

Reply:
I think option 2 is preferable.
But setting maxLength/padToMultipleOf when padding and truncation is disabled is harmless. I don't feel strong to print a warning to user. User might want to always set the `maxLength` and switch on/off padding

Reply:
I think this is fine as is. The situation I was thinking of is when neither truncation or padding are explicitly set, but user set the maxlength or pad to multiple of. It's harmless as you said, and i think it's fair to assume in this case the user knows what they are doing.